server/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -122,6 +163,34 @@ public Result editEligibility(Request request, long programId, long blockDefinit
   public Result updateVisibility(Request request, long programId, long blockDefinitionId) {
     requestChecker.throwIfProgramNotDraft(programId);
 
+    ReadOnlyQuestionService roQuestionService =
```

Comment:
You make this here but then only pass it to `generatePredicateDefinition()`.  Make it there where it's needed instead.

Reply:
It's consumed on line 254 as well